src/plugins/db2sql.c

```diff
@@ -811,6 +815,327 @@ void do_mobdb2sql(void)
+	}
+	// State
+	StrBuf->Printf(&buf, "'%s',", mob_skill_state_tostring(i32));
+
+	// SkillID
+	if (!(i32 = skill->name2id(name))) {
+		ShowWarning("mob_skill_db_libconfig_sub_skill: Non existant skill id %d in monster %d, skipping.\n", i32, mob_id);
+		return false;
+	}
+	StrBuf->Printf(&buf, "%d,", i32);
+
+	// SkillLv
+	if (!libconfig->setting_lookup_int(it, "SkillLevel", &i32) || i32 <= 0)
+		i32 = 1;
+	StrBuf->Printf(&buf, "%d,", i32);
+
+	// Rate
+	i32 = 0;
+	libconfig->setting_lookup_int(it, "Rate", &i32);
+	StrBuf->Printf(&buf, "%d,", i32);
+
+	// CastTime
+	i32 = 0;
+	libconfig->setting_lookup_int(it, "CastTime", &i32);
+	StrBuf->Printf(&buf, "%d,", i32);
+
+	// Delay
+	i32 = 0;
+	libconfig->setting_lookup_int(it, "Delay", &i32);
+	StrBuf->Printf(&buf, "%d,", i32);
+
+	// Cancelable
+	if (libconfig->setting_lookup_bool(it, "Cancelable", &i32)) {
+		StrBuf->Printf(&buf, "'%s',", ((i32 == 0) ? "no" : "yes"));
+	}
+
+	// Target
+	if (mob->lookup_const(it, "SkillTarget", &i32) && (i32 < MST_TARGET || i32 > MST_AROUND)) {
+		i32 = MST_TARGET;
+	}
+	StrBuf->Printf(&buf, "'%s',", mob_skill_target_tostring(i32));
+
+	// Condition
+	if (mob->lookup_const(it, "CastCondition", &i32) && (i32 < MSC_ALWAYS || i32 > MSC_SPAWN)) {
+		i32 = MSC_ALWAYS;
+	}
+	StrBuf->Printf(&buf, "'%s',", mob_skill_condition_tostring(i32));
+
+	// ConditionValue
+	i32 = 0;
+	mob->lookup_const(it, "ConditionData", &i32);
+	StrBuf->Printf(&buf, "'%d',", i32);
+
+	// Val1-Val5
+	for (i = 0; i < 5; i++) {
+		sprintf(valname, "val%1d", i);
+		if (libconfig->setting_lookup_int(it, valname, &i32)) {
+			StrBuf->Printf(&buf, "%d,", i32);
+		} else {
+			StrBuf->Printf(&buf, "NULL,");
+		}
+	}
+
+	// Emotion
+	if (libconfig->setting_lookup_int(it, "Emotion", &i32)) {
+		StrBuf->Printf(&buf, "%d,", i32);
+	} else {
+		StrBuf->Printf(&buf, "NULL,");
+	}
+
+	if (libconfig->setting_lookup_int(it, "ChatMsgID", &i32)) {
+		if (i32 > 0 && i32 <= MAX_MOB_CHAT) {
+			StrBuf->Printf(&buf, "%d", i32);
+		} else {
+			StrBuf->Printf(&buf, "NULL");
+		}
+	}
+
+	fprintf(tosql.fp, "REPLACE INTO `%s` VALUES (%s);\n", tosql.db_name, StrBuf->Value(&buf));
+
+	StrBuf->Destroy(&buf);
+
+	return true;
+
+}
+
+
+/**
+ * Prints a SQL table header for the current mob_skill_db table.
+ */
+void mobskilldb2sql_tableheader(void)
+{
+	db2sql_fileheader();
+
+	fprintf(tosql.fp,
+			"--\n"
+			"-- Table structure for table `%s`\n"
+			"--\n"
+			"\n"
+			"DROP TABLE IF EXISTS `%s`;\n"
+			"CREATE TABLE `%s` (\n"
+			"  `MOB_ID` SMALLINT(6) NOT NULL,\n"
+			"  `INFO` TEXT NOT NULL,\n"
+			"  `STATE` TEXT NOT NULL,\n"
+			"  `SKILL_ID` SMALLINT(6) NOT NULL,\n"
+			"  `SKILL_LV` TINYINT(4) NOT NULL,\n"
+			"  `RATE` SMALLINT(4) NOT NULL,\n"
+			"  `CASTTIME` MEDIUMINT(9) NOT NULL,\n"
+			"  `DELAY` INT(9) NOT NULL,\n"
+			"  `CANCELABLE` TEXT NOT NULL,\n"
+			"  `TARGET` TEXT NOT NULL,\n"
+			"  `CONDITION` TEXT NOT NULL,\n"
+			"  `CONDITION_VALUE` TEXT,\n"
+			"  `VAL1` MEDIUMINT(9) DEFAULT NULL,\n"
```

Comment:
probably for valX better use ``int(11)``? like other fields. Because if put item id in one of valX field, can be overflow. Now item id is full int32 values.

Reply:
Done.